avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeBase.java

```diff
@@ -109,6 +109,13 @@ protected Class compileClass(final String className, Set<String> knownUsedFullyQ
 
     JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
     String compileClassPathForCurrentFile = Utils.inferCompileDependencies(compileClassPath, filePath, knownUsedFullyQualifiedClassNameSet);
+    if (null == compiler) {
```

Comment:
worth moving this check one line up so that inferCompileDependencies
() isnt even called